### PR TITLE
MCR 3189 fix pmd issues in mycore solr

### PR DESCRIPTION
**Pull request by Antoniafriedrich**
[Link to jira](https://mycore.atlassian.net/browse/MCR-3189).

### CODE REVIEW

File: mycore-solr/src/main/java/org/mycore/solr/search/MCRSolrSearchUtils.java [hunk 1]
```diff
@@ -233,9 +236,11 @@ protected QueryResponse query(SolrParams params) {
         @Override
         public boolean tryAdvance(Consumer<? super SolrDocument> action) {
             if (action == null) {
-                throw new NullPointerException();
+                throw new IllegalArgumentException("Action cannot be null");
             }
-            long i = start, size = estimateSize();
+            Objects.requireNonNull(action);
```
Comment (rsteph-de) [inline mycore-solr/src/main/java/org/mycore/solr/search/MCRSolrSearchUtils.java]: will never fail after `if(action == null) throws`

File: mycore-solr/src/main/java/org/mycore/solr/index/handlers/content/MCRSolrMCRContentMapIndexHandler.java [hunk 2]
```diff
@@ -112,18 +115,17 @@ public void index() {
                         LOGGER.info("Sending {} documents was successful in {} ms.", totalCount,
                                 updateResponse.getElapsedTime());
                     }
-                } catch (Throwable e) {
+                } catch (SolrServerException|IOException e) {
```
Comment (rsteph-de) [inline mycore-solr/src/main/java/org/mycore/solr/index/handlers/content/MCRSolrMCRContentMapIndexHandler.java]: format code (space before+after `|`) ?

File: mycore-solr/src/main/java/org/mycore/solr/index/handlers/content/MCRSolrMCRContentMapIndexHandler.java [hunk 3]
```diff
@@ -112,18 +115,17 @@ public void index() {
...
                     LOGGER.warn("Error while indexing document collection. Split and retry.", e);
                     splitup(List.of(destinationCore));
                     return;
                 }
             }
-        } catch (Throwable e) {
+        } catch (SAXException | IOException e) {
             splitup(getDestinationCores());
-        } finally {
+        }  finally {
```
Comment (rsteph-de) [inline mycore-solr/src/main/java/org/mycore/solr/index/handlers/content/MCRSolrMCRContentMapIndexHandler.java]: format code - 2 spaces?

**Review by rsteph-de — CHANGES_REQUESTED** [review]
some minor issues, otherwise O.K.

**Review by toKrause — APPROVED** [review]

**Pull request merged**